### DISCUSSION TITLE
Nspanel built-in button delay (debounce?)

### DISCUSSION THREAD

**Discussion by peca89** [arendst/Tasmota#18941] [Q&A] [answered]
By default, after installing tasmota32-nspanel.bin, two internal buttons are linked to built-in relays which is very good for failsafe. However, I can notice significant delay between pressing a button and hearing the relay click. I'm not sure whether there is a software debounce time or if maybe software is waiting for the possible double click action, but 0.5-1s delay is difficult to accept on something that is intended to mimic the standard light switch. What also bothers me is that pressing both buttons at the same time (or pressing the second one during the debounce time of the first one) does not trigger the second one.

Are these actions handled by Tasmota rules? If so, how can I see them and modify them?
Is it possible to shorten the time delay?
Is it possible to make pressing both buttons at once work?
Is it possible to unlink these buttons and relays locally and hand the control over to something external via mqtt?

**Comment by sfromis** [comment, marked as answer]
While I'm not using the NSPanel, when you mention buttons and delays, I'm reminded of `SetOption13` allowing you to disable the extra features available with multi-press and hold, which only can work with a delay before determining that it was only a single press.

**Comment by peca89** [comment]
That's it, thank you!

SetOption32 looks like the one that sets the delay time. I'll check that out later.

**Reply by sfromis** [reply]
No, not SetOption32 which is only the delay for hold, default 4 seconds, not for multi-press detection. If what you want is "immediate action", SetOption13 is specifically for that, while SetOption32 only pertains to the much longer hold time. There is no similar option to control how fast the multi-press detection delay is.
